precompiles/gov/tx.go

```diff
@@ -33,7 +33,7 @@ func (p *Precompile) SubmitProposal(
 	method *abi.Method,
 	args []interface{},
 ) ([]byte, error) {
-	msg, proposerHexAddr, err := NewMsgSubmitProposal(args, p.codec)
+	msg, proposerHexAddr, err := NewMsgSubmitProposal(args, p.codec, p.addrCdc)
```

Comment:
Can we just make the `NewMsg*` functions members of the `Precompile` so they can just directly use the address codec from there?

Reply:
i feel message constructors should be functions, not methods.

Reply:
okey that is fair.  do the methods need to be public or can we make them private?

Reply:
hmm hard to say really - i'd wager it's not entirely out of the realm of possibility for a user to want to construct these messages for a different use case. maybe a custom module that does something with cosmos SDK messages. 🤷